### PR TITLE
Frontend Search Summary Query

### PR DESCRIPTION
**Pull request by profjsb**
This PR adds the ability to query the summary vector database with a natural language query, by adding a new frontend page `/summary_search`.
Kapture 2023-05-03 at 09 38 19

Changes the API from GET to POST and updates the tests.

TODO:
- [ ] Add front end tests for the new page

### CODE REVIEW

File: static/js/components/SummarySearch.jsx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+                <TableCell>Summary</TableCell>
+                <TableCell>Redshift</TableCell>
+                <TableCell>Classification(s)</TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {queryResult.query_results?.map((source) => {
+                let fw = "normal";
+                let col = "grey";
+                if (source.score > 0.9) {
```
Comment (profjsb) [inline static/js/components/SummarySearch.jsx:223]: scores above 0.9 appear like fairly good matches, colorize as such

File: static/js/components/SummarySearch.jsx [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+          {
+            properties: {
+              toggleA: { enum: [true] },
+              k: {
+                type: "integer",
+                title: "(Optional) Number of sources to return",
+                default: 5,
+                minimum: 1,
+                maximum: 25,
+                multipleOf: 1,
```
Comment (mcoughlin) [inline static/js/components/SummarySearch.jsx:122]: Needed?

File: static/js/components/SummarySearch.jsx [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+                <TableCell>Name</TableCell>
+                <TableCell>Score</TableCell>
+                <TableCell>Summary</TableCell>
+                <TableCell>Redshift</TableCell>
+                <TableCell>Classification(s)</TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {queryResult.query_results?.map((source) => {
+                let fw = "normal";
```
Comment (mcoughlin) [inline static/js/components/SummarySearch.jsx:275]: Should these be in the config?

**Review by mcoughlin — APPROVED** [review]
Small comments.

File: static/js/components/SummarySearch.jsx (hunk 2 above)
Comment (profjsb) [reply static/js/components/SummarySearch.jsx:122]: probably not

File: static/js/components/SummarySearch.jsx (hunk 3 above)
Comment (profjsb) [reply static/js/components/SummarySearch.jsx:277]: done

**Review by Theodlz — APPROVED** [review]
LGTM!

**Pull request merged**